File: 1206-corporate-flight-bookings/corporate-flight-bookings.py
class Solution:
    def corpFlightBookings(self, bookings: List[List[int]], n: int) -> List[int]:
        diff = [0] * (n + 2)
        for i in bookings:
            l, r, val = i
            diff[l] += val
            diff[r + 1] -= val
        # Line Sweep (All ranges are updated, find prefix)
        print(diff)
        prefix = []
        rs = 0
        for i in diff:
            rs += i
            prefix.append(rs)
        
        # Answer array (Addition of original[i] + prefix[i])
        ans = []
        for i in range(1, n + 1):
            ans.append(prefix[i])
        return ans

File: 1206-corporate-flight-bookings/corporate-flight-bookings.java
class Solution {
    public int[] corpFlightBookings(int[][] bookings, int n) {
        // Brute force, update all the flights from first to last using bookings
        int flights[] = new int[n]; // flights = [0] * n
        for (int[] booking: bookings) {
            int l = booking[0];
            int r = booking[1];
            int val = booking[2];
            for (int i = l - 1; i < r; i++) {
                flights[i] += val;
            }
        }
        return flights;
    }
}